### PR TITLE
Feat:Api timeout Configurable

### PR DESCRIPTION
**Pull request by Shivam-nagar23**
# Description
Setting the api timeout configurable through Config map.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Local
- [x] VM

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/userGroups/userGroup.service.ts [hunk 1]
```diff
@@ -62,11 +62,29 @@ export function getParsedData(teamList, appList, environmentList) {
 }
 
 export function saveUser(request: CreateUser) {
-    return request.id ? put(`user`, request) : post(`user`, request)
+    const options: APIOptions = {
+        timeout: window._env_.CONFIGURABLE_TIMEOUT ? parseInt(window._env_.CONFIGURABLE_TIMEOUT, 10) : 30,
```
Comment (milindk8) [inline src/components/userGroups/userGroup.service.ts:66]: Simply using if else condition & use `` only for string interpolation

File: src/components/userGroups/userGroup.service.ts [hunk 2]
```diff
@@ -62,11 +62,29 @@ export function getParsedData(teamList, appList, environmentList) {
...
+    }
+    return request.id
```
Comment (milindk8) [inline src/components/userGroups/userGroup.service.ts:68]: same here

File: src/components/userGroups/userGroup.service.ts [hunk 3]
```diff
@@ -62,11 +62,29 @@ export function getParsedData(teamList, appList, environmentList) {
...
+        : window._env_.CONFIGURABLE_TIMEOUT
+        ? post(`user`, request, options)
+        : post(`user`, request)
 }
 
 export function saveGroup(request: CreateGroup) {
-    return request.id ? put(`user/role/group`, request) : post(`user/role/group`, request)
+    const options: APIOptions = {
+        timeout: window._env_.CONFIGURABLE_TIMEOUT ? parseInt(window._env_.CONFIGURABLE_TIMEOUT, 10) : 30,
+    }
```
Comment (milindk8) [inline src/components/userGroups/userGroup.service.ts:80]: same

File: src/components/userGroups/userGroup.service.ts [hunk 4]
```diff
@@ -62,11 +62,28 @@ export function getParsedData(teamList, appList, environmentList) {
...
+
+    if (window._env_.CONFIGURABLE_TIMEOUT) {
+        console.log(options.timeout)
+        return request.id ? put('user/role/group', request, options) : post('user/role/group', request, options)
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/userGroups/userGroup.service.ts:83]: Seems code repetition in the `saveUser` and `saveGroup` functions. Please optimize  it

File: src/components/userGroups/userGroup.service.ts [hunk 5]
```diff
@@ -62,11 +62,28 @@ export function getParsedData(teamList, appList, environmentList) {
 }
 
 export function saveUser(request: CreateUser) {
-    return request.id ? put(`user`, request) : post(`user`, request)
+    const options: APIOptions = {
+        timeout: window._env_.CONFIGURABLE_TIMEOUT ? parseInt(window._env_.CONFIGURABLE_TIMEOUT, 10) : 30,
+    }
+    if (window._env_.CONFIGURABLE_TIMEOUT) {
+        console.log(options.timeout)
```
Comment (shivani170) [inline src/components/userGroups/userGroup.service.ts:69]: remove console

File: src/components/userGroups/userGroup.service.ts [hunk 6]
```diff
@@ -62,11 +62,28 @@ export function getParsedData(teamList, appList, environmentList) {
...
 }
 
 export function saveGroup(request: CreateGroup) {
-    return request.id ? put(`user/role/group`, request) : post(`user/role/group`, request)
+    const options: APIOptions = {
+        timeout: window._env_.CONFIGURABLE_TIMEOUT ? parseInt(window._env_.CONFIGURABLE_TIMEOUT, 10) : 30,
+    }
+
+    if (window._env_.CONFIGURABLE_TIMEOUT) {
+        console.log(options.timeout)
```
Comment (shivani170) [inline src/components/userGroups/userGroup.service.ts:82]: remove console

**Review by milindk8 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**